### ISSUE TITLE
Check startpage.com

### ISSUE THREAD

**Issue by probonopd** [Eloston/ungoogled-chromium#768]
Please consider using https://www.startpage.com/ as the default search engine. Its search results are closer to Google while still maintaining privacy.

**Comment by Eloston** [comment]
I've noticed searx has had issues with getting results from Google these days, so I wouldn't be against changing to a more functional search engine. (However, I don't know the current situation well so please let me know if I'm wrong)

At the very least, we could add Startpage as another option.

**Comment by probonopd** [comment]
To those who have voted down above: why? Have you tried Startpage and compared its result with the current default search engine?

**Comment by lipici** [comment]
i didnt voted down but i checked and its not so good : 
1) first five results are the same like any browser but the rest are not relevant 
2) search its the slowest from all browsers and i get boored until it shows
3) very hard to difference the sites from search results
4) text shown from search results its short

**Comment by probonopd** [comment]
My issue with duckduckgo is that its search results never seem to be as "good" as Google's

**Comment by lipici** [comment]
true but its a thing that i can pass by, most of the time we search for a mellody, a movie, a person, some news and it can manage.

**Comment by probonopd** [comment]
The current default, searx.me, is unusable for me, as I get:

![searx](https://user-images.githubusercontent.com/2480569/59564217-4ddcc280-9044-11e9-820f-0b2c22c6d73c.png)

startpage.com is much more reliable for me.

**Comment by probonopd** [comment]
> At the very least, we could add Startpage as another option.

It is there as another option already, I just believe it should be the default.

**Comment by Eloston** [comment]
@probonopd It's not included by default in a fresh profile.

**Comment by theel0ja** [comment]
I don't think default search engine should be set (like it's now), but I think Startpage should be added as an option.

**Comment by emanruse** [comment]
Startpage now requires JS. Nothing that tells you MUST use JS can be considered to maintain privacy. Also DuckDuckGo is hosted on Amazon, so their claim of privacy is nonsense.

> searx.me

There are also other searx instances:
https://github.com/asciimoo/searx/wiki/Searx-instances

**Comment by theel0ja** [comment]
>Startpage now requires JS.

Wrong. It works just fine with Tor Browser "Safest" setting (disables JS/SVG/etc, to find: click that :shield: icon in the toolbar)

**Comment by emanruse** [comment]
> Wrong. It works just fine with Tor Browser "Safest" setting (disables JS/SVG/etc, to find: click that :shield: icon in the toolbar)

And the moment you try to search for something that "fine" is turns into a mess with no search results.

Additionally at least 2 requests get blocked by uBO which are blacklisted in EasyPrivacy list. So you can figure how "most private" that search engine is.

**Comment by theel0ja** [comment]
>And the moment you try to search for something that "fine" is turns into a mess with no search results.

It works without problems for me.

**Comment by jlj2** [comment]
A preferred setting for privacy when using startpage would arguably be to 'Use POST vs GET'  in the 'Privacy & Safety [settings webpage](https://www.startpage.com/search/settings), but then the obfuscated URL that you could use (see bottom of page)  could not be added as a search engine in chromium (nor ungoogled-chromium) under ['Manage search engines'](chrome://settings/searchEngines?search=search) in my experience, as the url does not contain an _%s_ for the search term.
A workable alternative is to launch that search engine's webpage when required from a bookmark, say, in the bookmarks bar.

**Comment by emanruse** [comment]
Why bother with startpage.com at all when you have many searx instances (including .onion ones)? If you so insist on seeing results from startpage only, just use the search operator "!startpage" - it will show you that. Similarly you can search "!duckduckgo" etc.

Re. GET/POST:

Well, chromium's requirement about %s seems really restrictive in regards to that. So the best thing (privacy wise and anti-mistake wise) is to use 
http://127.0.0.1/%s as the default search engine and have a searx instance bookmarked for easy access. (Yes, I know it will need one additional click, what a nightmare :P)

**Comment by theel0ja** [comment]
>A preferred setting for privacy when using startpage would arguably be to 'Use POST vs GET'

The difference there is that it doesn't save the query to browsing history, which may not be enabled.

>Why bother with startpage.com at all when you have many searx instances

They often suck in usability.

https://searx.me/?q=Wikipedia&categories=general&language=en-US returns whole 1 result and an error message

For example https://searx.dnswarden.com/?q=Wikipedia&categories=general&language=en-US seems to be better, though.

Also I think Startpage should be added as option, but not as default (no default at the moment, so user choice)

**Comment by jstkdng** [comment]
I read somewhere that startpage and google made a deal so that startpage can use google results as long as google gets the ip from the person who is making the request. If that's the case, then I would strongly suggest against it, could be wrong though.

**Comment by emanruse** [comment]
> >Why bother with startpage.com at all when you have many searx instances  
> 
> They often suck in usability.
> 
> https://searx.me/?q=...

searx.me is #54 in the instances classification:

https://stats.searx.xyz/

**Comment by csagan5** [comment]
There's been a [similar discussion for Bromite](https://github.com/bromite/bromite/issues/91) and I have been against adding Searx for a reason which has nothing to do with usability: we need to find who is accountable for the logs anda PII, and what is at stake for them to compromise them (e.g. security level they can afford and implement).

**Comment by PF4Public** [comment]
This is an old issue, that didn't show much activity recently — closing.
If you have any more information to add, let us know.

**Comment by csagan5** [comment]
Slightly related: there is a bug in upstream Chromium that prevents a lot of search engines from appearing even if they have a valid OpenSearch descriptor.

In next release of Bromite (`100.0.4896.83`) I will add the bugfix to [Bromite's cumulative OpenSearch patch](https://github.com/bromite/bromite/blob/master/build/patches/OpenSearch-allow-paths-and-incognito-mode.patch), you will find this link working when release is live: https://github.com/bromite/bromite/blob/100.0.4896.83/build/patches/OpenSearch-allow-paths-and-incognito-mode.patch

**Comment by PF4Public** [comment]
Hmm… Seems like bromite has some new patches, unseen to ungoogled-chromium. I think, I'll open another issue to discuss if some of them might be incorporated here.

@csagan5 thanks for mentioning this patch. Also if you by any chance could name those of bromite patches which are applicable and useable for ungoogled-chromium, that'd be great.

**Comment by csagan5** [comment]
> Also if you by any chance could name those of bromite patches which are applicable and useable for ungoogled-chromium, that'd be great.

You could check the [CHANGELOG](https://github.com/bromite/bromite/blob/master/CHANGELOG.md) for applicable improvements.

**Comment by csagan5** [comment]
I pre-released the patch is here so that others can see/use it before next Bromite release: https://github.com/bromite/bromite/blob/b4095e087a3aad4e1e6c4d3a890873b1cdd5fed1/build/patches/OpenSearch-allow-paths-and-incognito-mode.patch

With the introdution of it I can see that a lot more search engines show up as working first time their page is visited. Someone should upstream the fix in Chromium as well.
